src/Sylius/Bundle/CoreBundle/Cart/ItemResolver.php

```diff
@@ -127,7 +127,7 @@ public function resolve(CartItemInterface $item, $data)
 
         // We use forms to easily set the quantity and pick variant but you can do here whatever is required to create the item.
         $form = $this->formFactory->create('sylius_cart_item', $item, array('product' => $product));
-        $form->submit($data);
+        $form->handleRequest($data);
```

Comment:
very true, but i'd rather keep this PRs changes specific to fixing deprecations.
that can be done in another PR